sympy/vector/coordsyssph.py

```diff
@@ -0,0 +1,211 @@
+                           (r'\mathbf{\hat{θˆ}_{%s}}' % name),
+                           (r'\mathbf{\hat{φˆ}_{%s}}' % name)]
+            pretty_vects = (name + '_rˆ', name + '_θˆ', name + '_φˆ')
+        else:
+            _check_strings('vector_names', vector_names)
+            vector_names = list(vector_names)
+            latex_vects = [(r'\mathbf{\hat{%s}_{%s}}' % (x, name)) for
+                           x in vector_names]
+            pretty_vects = [(name + '_' + x) for x in vector_names]
+
+        obj._rˆ = BaseVector(vector_names[0], 0, obj,
+                            pretty_vects[0], latex_vects[0])
+        obj._θˆ = BaseVector(vector_names[1], 1, obj,
+                            pretty_vects[1], latex_vects[1])
+        obj._φˆ = BaseVector(vector_names[2], 2, obj,
+                            pretty_vects[2], latex_vects[2])
+
+        # Initialize the base scalars
+        if variable_names is None:
+            variable_names = (name + '.r', name + '.θ', name + '.φ')
+            latex_scalars = [(r"\mathbf{{r}_{%s}}" % name),
+                             (r"\mathbf{{θ}_{%s}}" % name),
+                             (r"\mathbf{{φ}_{%s}}" % name)]
+            pretty_scalars = (name + '_r', name + '_θ', name + '_φ')
+        else:
+            _check_strings('variable_names', vector_names)
+            variable_names = list(variable_names)
+            latex_scalars = [(r"\mathbf{{%s}_{%s}}" % (x, name)) for
+                             x in variable_names]
+            pretty_scalars = [(name + '_' + x) for x in variable_names]
+
+        obj._r = BaseScalar(variable_names[0], 0, obj,
+                            pretty_scalars[0], latex_scalars[0])
+        obj._θ = BaseScalar(variable_names[1], 1, obj,
+                            pretty_scalars[1], latex_scalars[1])
+        obj._φ = BaseScalar(variable_names[2], 2, obj,
+                            pretty_scalars[2], latex_scalars[2])
+
+
+        # Assign a Del operator instance
+        from sympy.vector.deloperator import Del
+        obj._delop = Del(obj)
+
+        # Assign params
+        obj._parent = parent
+        if obj._parent is not None:
+            obj._root = obj._parent._root
+        else:
+            obj._root = obj
+
+        obj._parent_rotation_matrix = parent_orient
+        obj._origin = origin
+
+        # Return the instance
+        return obj
+
+    def __str__(self, printer=None):
+        return self._name
+
+    __repr__ = __str__
+    _sympystr = __str__
+
+    def __iter__(self):
+        return iter([self.i, self.j, self.k])
+
+    @property
+    def origin(self):
+        return self._origin
+
+    @property
+    def delop(self):
+        return self._delop
+
+    @property
+    def rˆ(self):
+        return self._rˆ
+
+    @property
+    def θˆ(self):
+        return self._θˆ
+
+    @property
+    def φˆ(self):
+        return self._φˆ
+
+    @property
+    def r(self):
+        return self._r
+
+    @property
+    def θ(self):
```

Comment:
These should be ascii names so that you can easily type them into the interpreter and scripts.